EC2 Route

Dkaykay opened the pull request:
### Description of your changes

Relates to #657 

Implemented Route 

I have:

- [ x ] Read and followed Crossplane's [contribution process].
- [ x ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

- create route
- delete route
- run make reviewable test

haarchri commented on examples/ec2/route.yaml:
```diff
@@ -0,0 +1,12 @@
+  forProvider:
+    destinationCIDRBlock: 10.10.10.0/24
+    region: eu-central-1
+    routeTableID: rtb-00000000000000001
```
can we add here referencer ? 
reference resolver generation got recently merged - https://github.com/crossplane/crossplane-tools#reference-resolvers if you want to give a try

Dkaykay replied:
Hi @haarchri,
i talked to @AaronME these days, we had some issues with this PR.
We spent a lot of time with this and decided to first merge and add further features at a later point when we have more free time. 
I tried to use the generator for creating the referencers, but it did not work out of the box, so i dont want to loose more time for now, my team really needs this feature. The initial PR was already created months ago.

haarchri commented on examples/ec2/route.yaml:
```diff
@@ -0,0 +1,12 @@
+    destinationCIDRBlock: 10.10.10.0/24
+    region: eu-central-1
+    routeTableID: rtb-00000000000000001
+    vpcPeeringConnectionID: pcx-00000000000000001
```
can we add here referencer ? 
reference resolver generation got recently merged - https://github.com/crossplane/crossplane-tools#reference-resolvers if you want to give a try

haarchri commented:
@Dkaykay @AaronME  looks like i have the same issue after rebasing master for https://github.com/crossplane/provider-aws/pull/831  https://github.com/crossplane/provider-aws/issues/876

```
github.com/crossplane/provider-aws/apis/ec2/v1alpha1:-: CRD for Instance.ec2.aws.crossplane.io has no storage version
Error: not all generators ran successfully
```

Dkaykay commented:
still blocked by #920

haarchri commented:
@Dkaykay #920  is merged so you can give this PR a new try

haarchri commented:
@Dkaykay  i added a new commit:
rebased & added create, observe, delete & resolvers

provider-aws currently provides both a standalone Route (with this PR) resource and a RouteTable (in v1beta1) resource with routes defined in-line. At this time you cannot use a Route Table with in-line routes in conjunction with any Route resources. Doing so will cause a conflict of rule settings and will overwrite rules. Thats why i decided not create a resolver for `routeTableID`


```
kubectl get managed
NAME                                  READY   SYNCED   EXTERNAL-NAME
route.ec2.aws.crossplane.io/example   True    True     example

```

Dkaykay commented:
> @Dkaykay i added a new commit: rebased & added create, observe, delete & resolvers
> 
> provider-aws currently provides both a standalone Route (with this PR) resource and a RouteTable (in v1beta1) resource with routes defined in-line. At this time you cannot use a Route Table with in-line routes in conjunction with any Route resources. Doing so will cause a conflict of rule settings and will overwrite rules. Thats why i decided not create a resolver for `routeTableID`
> 
> ```
> kubectl get managed
> NAME                                  READY   SYNCED   EXTERNAL-NAME
> route.ec2.aws.crossplane.io/example   True    True     example
> ```

Hey @haarchri, thats awesome! Thanks you very much 👍🏼
I will have a discussion on this PR with @AaronME today :)

AaronME reviewed (approved):
Creates and destroys routes. LGTM! Thank you for your patience and hard work @Dkaykay!

I've opened https://github.com/crossplane/provider-aws/issues/1026 to address the references and selectors.
